### DISCUSSION TITLE
What would a porter deployment of MongoDB look like?

### DISCUSSION THREAD

**Discussion by Steve973** [getporter/porter#2511] [Q&A]
Hello.  It has been a while since I have looked into Porter, so I wanted to check back in and see how things are going.  I have been using Docker, Podman, and lately, I have begun to learn some of kubernetes via k3s.  I often have to start up containers of things like MongoDB with a host filesystem directory to mount for the database (files) directory.  Currently, I have a bunch of bash scripts that create the directories, and prepare everything before I start up the various containers.  I would love to streamline this and use something like Porter, but I am wondering if this is something that Porter can handle.  If so, how would this work, or what would it look like?  I get that this is "cloud native", but data needs to live concretely in one (or many) locations, depending on the type of data store, and things like sharding or replication.

If it would help to tie this into something a bit more concrete, I have a github repo that I have been using to build what I'm learning.  It has a bunch of k3s yaml files that define services, deployments, and ingress routes for several data applications.  It also contains a helm chart config to customize the startup of the Traefik proxy that k3s uses as its ingress router.  I also have the equivalent stack set up to run with podman (non-root).  So, if it would be helpful to tie this discussion to something concrete, I do not mind sharing a link to my repo.  Just let me know, and thanks for your time.

**Comment by carolynvs** [comment]
Hey @Steve973, sorry for the slow reply, we are all still coming back from vacation.

I am extremely interested in your repo, it sounds like a really common (and currently unsupported) way to use Porter that would be useful to lots of people, so please post a link. 👍 

We have a pull request (#2167) that implements mounting local volumes into the bundle which seems to be the gap for making your scenario. I need some time to review it and get it merged but it will be merged.

**Reply by Steve973** [reply]
Hi, @carolynvs and I apologize for taking this long to get back to you.  Let me clarify something that I mentioned: the personal repository that I referenced does not attempt to lay down directories on the host.  It merely sets up MongoDB to run, and I would like to figure out how to use Porter to lay down resource directories as part of the process of provisioning the host node(s) of an app that needs persistent storage volumes.  If this is something that isn't particularly difficult, then that shows how new I am to kubernetes, cloud deployments, and CNAB.  So, please excuse my ignorance if have overlooked something routine, easy, or obvious.

With that clarification out of the way, is there anything to be done?  If so, how can I help?
